openml/extensions/extension_interface.py

```diff
@@ -0,0 +1,91 @@
+from abc import ABCMeta, abstractmethod
+from typing import Any, List, Optional, Tuple
+
+from openml import OpenMLTask, OpenMLFlow
+from openml.runs.trace import OpenMLTraceIteration, OpenMLRunTrace
+
+
+class Extension:
+    __metaclass__ = ABCMeta
```

Comment:
[PEP3115](https://www.python.org/dev/peps/pep-3115/) switches the syntax for adding a meta-class with Py3 to `class Extension(metaclass=ABCMeta):`, see also [updated docs](https://docs.python.org/3/library/abc.html#abc.ABC).

Reply:
It even seems advised to subclass `abc.ABC` directly instead of specifying the metaclass from a legibility point [see notes](https://docs.python.org/3/whatsnew/3.4.html#abc). Eitherway works for me.